python/ray/serve/controller.py

```diff
@@ -490,26 +495,31 @@ def deploy_apps(
         # deprecate such usage.
         if isinstance(config, ServeApplicationSchema):
             config = config.to_deploy_schema()
+            if self.deploy_mode == ServeDeployMode.MULTI_APP:
+                raise RayServeException(
+                    "You are trying to deploy a single-application config, however "
+                    "a multi-application config has been deployed to the current "
+                    "Serve instance already. Mixing single-app and multi-app is not "
+                    "allowed. Please either redeploy using the multi-application "
+                    "config format `ServeDeploySchema`, or shutdown and restart Serve "
+                    "to submit a single-app config of format `ServeApplicationSchema`. "
+                    "If you are using the REST API, you can submit a single-app config "
+                    "to the single-app API endpoint `/api/serve/deployments/`."
+                )
+            self.deploy_mode = ServeDeployMode.SINGLE_APP
         else:
-            # TODO (zcin): ServeApplicationSchema still needs to have host and port
-            # fields to support single-app mode, but in multi-app mode the host and port
-            # fields at the top-level deploy config is used instead. Eventually, after
-            # migration, we should remove these fields from ServeApplicationSchema.
-            host, port = config.host, config.port
-            for app_config in config.applications:
-                app_config_dict = app_config.dict(exclude_unset=True)
-                if "host" in app_config_dict:
-                    logger.info(
-                        f"Host {app_config_dict['host']} is set in the config for "
-                        f"application `{app_config.name}`. This will be ignored, as "
-                        f"the host {host} from the top level deploy config is used."
-                    )
-                if "port" in app_config:
-                    logger.info(
-                        f"Port {app_config_dict['port']} is set in the config for "
-                        f"application `{app_config.name}`. This will be ignored, as "
-                        f"the port {port} from the top level deploy config is used."
-                    )
```

Comment:
Removed the warnings here, as in a separate PR I will add validation in the schema itself.